File: Data Structures & Algorithms/max-water-container/submission-7.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        i, j = 0, len(heights) - 1
        res = 0
        while i < j:
            current = min(heights[i], heights[j]) * (j - i)
            if heights[i] < heights[j]:
                i += 1
            else:
                j -= 1
            res = max(current, res)
        return res

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-7.rs
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut profit = 0;
        let mut minBuy = &prices[0];
        for p in &prices {
            minBuy = min(&p, minBuy);
            profit = max(p - minBuy, profit);
        }
        profit
    }
}
